saml.Test/saml.Test/TestResponseClassSaml.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Saml;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+using System.IO;
+using System;
+using System.Xml;
+
+namespace Saml.Test
+{
+    [TestClass]
+    public class TestResponseClassSaml
+    {
+        /// <summary>
+        /// This test method ensures that the getter functions return the correct information.
+        /// </summary>
+        [TestMethod()]
+        public void testGetters()
+        {
+            Response response = new Response(Constants.VALID_CERTIFICATE);
+
+            string xml_contents = getResourceContents(Constants.VALID_XML_RESPONSE_RESOURCE);
+
+            response.LoadXml(xml_contents);
+
+            string display_name     = response.GetDisplayName();
+            string first_name       = response.GetFirstName();
+            string surname          = response.GetLastName();
+            string email_address    = response.GetEmail();
+            string name_id          = response.GetNameID();
+            string department       = response.GetDepartment();
+            string homephone        = response.GetPhone();
+            string company_name     = response.GetCompany();
+
+            List<string> groups     = response.GetGroups();
+
+            Assert.AreEqual("George Willy", display_name);
+            Assert.AreEqual("George", first_name);
+            Assert.AreEqual("Willy", surname);
+            Assert.AreEqual("george@george.com", email_address);
+            Assert.AreEqual("george@george.com", name_id);
+            Assert.AreEqual("Car repairs", department);
+            Assert.AreEqual("04567843543", homephone);
+            Assert.AreEqual("Displayr", company_name);
+
+            Assert.AreEqual("12345678-testing1-testing2", groups[0]);
+            Assert.AreEqual("87654321-testing4-testing3", groups[1]);
+        }
+
+        /// <summary>This test method loads an empty XML SAML response and a legitimate certificate
+        /// into the 'Response' class of the SAML library. The test ensures that the appropriate 
+        /// exception is thrown.</summary>
+        [TestMethod()]
+        public void testEmptyXml()
+        {
+            Response response = null;
```

Comment:
Minor thing but you can combine the definition and assignment into one line, i.e.

`Response response = new Response(...)`

Reply:
Again this is in other places too.